contracts/utils/Strings.sol

```diff
@@ -113,4 +128,261 @@ library Strings {
+     */
+    function parseUint(string memory input, uint256 begin, uint256 end) internal pure returns (uint256) {
+        (bool success, uint256 value) = tryParseUint(input, begin, end);
+        if (!success) revert StringsInvalidChar();
+        return value;
+    }
+
+    /**
+     * @dev Variant of {parseUint-string} that returns false if the parsing fails because of an invalid character.
+     *
+     * Requirements:
+     * - The result must fit into an `uint256` type.
+     */
+    function tryParseUint(string memory input) internal pure returns (bool success, uint256 value) {
+        return tryParseUint(input, 0, bytes(input).length);
+    }
+
+    /**
+     * @dev Variant of {parseUint-string-uint256-uint256} that returns false if the parsing fails because of an invalid
+     * character.
+     *
+     * Requirements:
+     * - The result must fit into an `uint256` type.
+     */
+    function tryParseUint(
+        string memory input,
+        uint256 begin,
+        uint256 end
+    ) internal pure returns (bool success, uint256 value) {
+        bytes memory buffer = bytes(input);
+
+        uint256 result = 0;
+        for (uint256 i = begin; i < end; ++i) {
+            uint8 chr = _tryParseChr(buffer[i]);
+            if (chr > 9) return (false, 0);
+            result *= 10;
+            result += chr;
+        }
+        return (true, result);
+    }
+
+    /**
+     * @dev Parse a decimal string and returns the value as a `int256`.
+     *
+     * Requirements:
+     * - The string must be formatted as `[-+]?[0-9]*`
+     * - The result must fit in an `int256` type.
+     */
+    function parseInt(string memory input) internal pure returns (int256) {
+        return parseInt(input, 0, bytes(input).length);
+    }
+
+    /**
+     * @dev Variant of {parseInt-string} that parses a substring of `input` located between position `begin` (included) and
+     * `end` (excluded).
+     *
+     * Requirements:
+     * - The substring must be formatted as `[-+]?[0-9]*`
+     * - The result must fit in an `int256` type.
+     */
+    function parseInt(string memory input, uint256 begin, uint256 end) internal pure returns (int256) {
+        (bool success, int256 value) = tryParseInt(input, begin, end);
+        if (!success) revert StringsInvalidChar();
+        return value;
+    }
+
+    /**
+     * @dev Variant of {parseInt-string} that returns false if the parsing fails because of an invalid character.
+     *
+     * Requirements:
+     * - The result must fit into an `int256` type.
+     */
+    function tryParseInt(string memory input) internal pure returns (bool success, int256 value) {
+        return tryParseInt(input, 0, bytes(input).length);
+    }
+
+    /**
+     * @dev Variant of {parseInt-string-uint256-uint256} that returns false if the parsing fails because of an invalid
+     * character.
+     *
+     * This function will still revert if the result does not fit in a `int256`
```

Comment:
Since this is a `try` function, shouldn't it return `success == false`?

Reply:
The revert is caused by 
```solidity
result *= 10;
result += factor * int8(chr);
```
overflowing.

Note that this is done in a loop. Any branch that you add here will be multiplied by the number of characteres in the substring. 

If you can find a nice and cheap way to return false, please feel free to change that

Reply:
doesn't the revert also happen for `tryParseUint`?

Reply:
Yes, it affect all variants (3) of this function. (uint, int, hex)

Reply:
Thanks all for your comments. I think the contract is currently well-documented anyway, and updating to a version that doesn't revert (in the future) is the kind of breaking change that'd be fine in a minor version.

I wouldn't block merging

Reply:
Alternatively, we can check that the length of the string in `tryParseUint` is >= 78, and check for overflow use `Math.tryAdd` in the last digit if the string's length == 78. (78 digits are in 2*256)

Similarly for `tryParseHex` but with 64 digits instead.

Reply:
@ernestognw zero prefix would create false positives. I'm not sure we can rule them out.